### PR TITLE
Refactor package to get it releasable 

### PR DESCRIPTION
**Pull request by nabobalis**
Updated instr from master as well.

### CODE REVIEW

**Comment by nabobalis** [comment]
Need to update all the docstrings.

**Comment by nabobalis** [comment]
Broken the tests now.

**Comment by nabobalis** [comment]
Windows and oldest deps now failing with:
```
E   AttributeError: type object 'pandas._libs.interval.IntervalMixin' has no attribute '__reduce_cython__'
```
Online tests failing with a hash issue.

**Comment by nabobalis** [comment]
I cannot replicate this locally.

File: sunkit_instruments/lyra/lyra.py [hunk 1]
```diff
@@ -110,7 +108,8 @@ def remove_lytaf_events_from_timeseries(ts, artifacts=None,
     """
     # Check that input argument is of correct type
     if lytaf_path:
-        warn('laytaf_path is deprecated, has no effect and will be removed in SunPy 2.1.', SunpyDeprecationWarning)
+        warn('lytaf_path is deprecated, has no effect and will be removed in SunPy 2.1.',
```
Comment (nabobalis) [inline sunkit_instruments/lyra/lyra.py:111]: Time to remove this?

**Comment by dstansby** [comment]
Ignoring the `numpy.ufunc size changed, may indicate binary incompatibility` warning in `setup.cfg` will fix some of the test collection

**Comment by nabobalis** [comment]
> Ignoring the `numpy.ufunc size changed, may indicate binary incompatibility` warning in `setup.cfg` will fix some of the test collection

Should we be ignoring that?

**Comment by dstansby** [comment]
🤷 I've never worked out how to actually fix it and ignore it in a bunch of projects (sunpy core included)

**Comment by nabobalis** [comment]
Oh we ignore it in sunpy? I had no idea.

File: setup.cfg [hunk 2]
```diff
@@ -1,60 +1,139 @@
 [metadata]
 name = sunkit_instruments
-author = The SunPy Developers
+provides = sunkit_instruments
+description = "A sunpy-affiliated package for solar instrument-specific tools."
```
Comment (wtbarnes) [inline setup.cfg:4]:
```suggestion
description = "A SunPy-affiliated package for solar instrument-specific tools."
```

It's the project name so it should be capitalized I think?

File: sunkit_instruments/__init__.py [hunk 3]
```diff
@@ -1,7 +1,33 @@
-# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+Sunkit-instruments
+==================
+
+A sunpy-affiliated package for solar instrument-specific tools.
```
Comment (wtbarnes) [inline sunkit_instruments/__init__.py:5]:
```suggestion
A SunPy-affiliated package for solar instrument-specific tools.
```

File: sunkit_instruments/lyra/lyra.py [hunk 4]
```diff
@@ -14,50 +12,43 @@
...
                                         force_use_local_lytaf=False):
     """
     Removes periods of LYRA artifacts defined in LYTAF from a TimeSeries.
 
     Parameters
     ----------
     ts : `sunpy.timeseries.TimeSeries`
-
-    artifacts : list of strings
+    artifacts : list
```
Comment (wtbarnes) [inline sunkit_instruments/lyra/lyra.py:43]:
```suggestion
    artifacts : `list`
```

**Review by wtbarnes — APPROVED** [review]
Just a few minor things. LGTM

File: pyproject.toml [hunk 5]
```diff
@@ -1,10 +1,35 @@
 [build-system]
+requires = [
+         "setuptools",
+         "setuptools_scm",
+         "wheel",
+         "oldest-supported-numpy",
```
Comment (Cadair) [inline pyproject.toml:6]: Do we actually have a build time requirement on numpy?

File: pyproject.toml [hunk 6]
```diff
@@ -1,10 +1,35 @@
...
+         "extension-helpers"
```
Comment (Cadair) [inline pyproject.toml:7]: Are we actually using anything from extension helpers?

File: pyproject.toml (hunk 5 above)
Comment (nabobalis) [reply pyproject.toml:6]: No.

File: pyproject.toml (hunk 6 above)
Comment (nabobalis) [reply pyproject.toml:7]: No.

File: setup.py [hunk 7]
```diff
@@ -1,28 +1,49 @@
...
 import os
+import sys
 from itertools import chain
 
-from setuptools import setup
 from setuptools.config import read_configuration
 
 ################################################################################
-# Programmatically generate some extras combos.
+# Raise helpful messages for test and build_docs commands
```
Comment (Cadair) [inline setup.py:10]: We don't really need this in a new package? tbh could probably pull it from core now as well

Comment (nabobalis) [reply setup.py:10]: Could remove it.

File: pyproject.toml (hunk 5 above)
Comment (Cadair) [reply pyproject.toml:6]: delete it!!

**Comment by nabobalis** [comment]
Seeing intermittent fails online:
```
FAILED ../../.tox/py38-online/lib/python3.8/site-packages/sunkit_instruments/goes_xrs/tests/test_goes_xrs.py::test_goes_chianti_tem_case2 - ValueError: ['http://hesperia.gsfc.na...
FAILED ../../.tox/py38-online/lib/python3.8/site-packages/sunkit_instruments/goes_xrs/tests/test_goes_xrs.py::test_goes_chianti_tem_case4 - ValueError: ['http://hesperia.gsfc.na...
FAILED ../../.tox/py38-online/lib/python3.8/site-packages/sunkit_instruments/goes_xrs/tests/test_goes_xrs.py::test_goes_chianti_tem_case5 - ValueError: ['http://hesperia.gsfc.na...
```

File: pyproject.toml (hunk 5 above)
Comment (nabobalis) [reply pyproject.toml]: Done.

File: pyproject.toml (hunk 6 above)
Comment (nabobalis) [reply pyproject.toml]: Done.

File: setup.py (hunk 7 above)
Comment (nabobalis) [reply setup.py]: Done.

**Comment by nabobalis** [comment]
> Seeing intermittent fails online:
> 
> ```
> FAILED ../../.tox/py38-online/lib/python3.8/site-packages/sunkit_instruments/goes_xrs/tests/test_goes_xrs.py::test_goes_chianti_tem_case2 - ValueError: ['http://hesperia.gsfc.na...
> FAILED ../../.tox/py38-online/lib/python3.8/site-packages/sunkit_instruments/goes_xrs/tests/test_goes_xrs.py::test_goes_chianti_tem_case4 - ValueError: ['http://hesperia.gsfc.na...
> FAILED ../../.tox/py38-online/lib/python3.8/site-packages/sunkit_instruments/goes_xrs/tests/test_goes_xrs.py::test_goes_chianti_tem_case5 - ValueError: ['http://hesperia.gsfc.na...
> ```
> 
> Somehow the hashes are wrong or not found?

Seems to change
```python
FAILED ../../.tox/py38-online/lib/python3.8/site-packages/sunkit_instruments/goes_xrs/tests/test_goes_xrs.py::test_calc_rad_loss_nokwags - ValueError: ['http://hesperia.gsfc.nas...
FAILED ../../.tox/py38-online/lib/python3.8/site-packages/sunkit_instruments/goes_xrs/tests/test_goes_xrs.py::test_calc_rad_loss_obstime - ValueError: ['http://hesperia.gsfc.nas...
```
anyone any ideas?

**Comment by nabobalis** [comment]
Seems intermittent which isn't good.

**Pull request merged**